naloge/kocke.tex

```diff
@@ -12,6 +12,50 @@
 
 Kako bi nalogo rešil, če je dovoljena tudi rotacija kock?
 \end{vprasanje}
+
 \begin{odgovor}
+Naj bo $v_i$ višina najvišjega stolpa, v katerem je na vrhu $i$-ta kocka.
+Zapišimo začetne pogoje in rekurzivne enačbe.
+\begin{align*}
+a_0 &= b_0 = \infty \\
+v_0 &= 0 \\
+v_i &= \max\set{v_j + c_i}{0 \le j \le i-1, a_i < a_j, b_i < b_j}
+\quad (1 \le i \le n)
+\end{align*}
+Vrednosti $v_i$ računamo naraščajoče po indeksu $i$ $(1 \le i \le n)$.
+Največjo višino dobimo kot $v^* = \max\set{v_i}{1 \le i \le n}$.
+
+\medskip
+
+V primeru, ko dovolimo rotacije,
```

Comment:
Ta rešitev ne deluje v splošnem, saj upošteva le en možen vrstni red kock - lahko bi preverili vse možne vrstne rede, a bi potem algoritem tekel v eksponentnem času. Tudi pristop z DAG ne deluje, saj bi se lahko posamezna kocka v rešitvi pojavila dvakrat - če se želimo temu izogniti, bi morali beležiti delne rešitve za vse kombinacije dovoljenih kock, kar bi spet prineslo eksponentno zahtevnost.

Tako bom iz naloge odstranil vprašanje z dovoljenimi rotacijami in to rešitev.